staging/maven-plugin/src/main/java/org/sonatype/nexus/maven/staging/deploy/strategy/StagingDeployStrategy.java

```diff
@@ -106,60 +165,93 @@ public void finalizeDeploy(final FinalizeDeployRequest request)
-    final List<StagingRepository> zappedStagingRepositories = new ArrayList<StagingRepository>();
-    for (File profileDirectory : localStageRepositories) {
-      if (!profileDirectory.isDirectory()) {
+
+    final Map<String, List<StagingRepository>> zappedStagingRepositories = Maps.newHashMap();
+    for (File nexusUrlFolder : nexusUrlFolders)
+    {
+      if (!nexusUrlFolder.isDirectory())
+      {
         continue;
       }
 
-      // we do remote staging
-      final String profileId = profileDirectory.getName();
-      log.info("");
-      log.info(" * Remote staging into staging profile ID \"{}\"", profileId);
-
-      try {
-        final Profile stagingProfile = remoteNexus.getStagingWorkflowV2Service().selectProfile(profileId);
-        final StagingRepository stagingRepository = beforeUpload(request.getParameters(), remoteNexus, stagingProfile);
-        zappedStagingRepositories.add(stagingRepository);
-        log.info(" * Uploading locally staged artifacts to profile {}", stagingProfile.name());
-        deployUp(request.getMavenSession(), getStagingDirectory(request.getParameters().getStagingDirectoryRoot(),
-            profileId),
-            createDeploymentArtifactRepository(remoteNexus.getServer().getId(), stagingRepository.getUrl()));
-        log.info(" * Upload of locally staged artifacts finished.");
-        afterUpload(request.getParameters(), remoteNexus, stagingRepository);
-      }
-      catch (NexusClientNotFoundException e) {
-        afterUploadFailure(request.getParameters(), remoteNexus, zappedStagingRepositories, e);
-        log.error("Remote staging finished with a failure: {}", e.getMessage());
-        log.error("");
-        log.error("Possible causes of 404 Not Found error:");
-        log.error(
-            " * your local workspace is \"dirty\" with previous runs, that locally staged artifacts? Run \"mvn clean\"...");
-        log.error(
-            " * remote Nexus got the profile with ID \"{}\" removed during this build? Get to Nexus admin...",
-            profileId);
-        throw new ArtifactDeploymentException("Remote staging failed: " + e.getMessage(), e);
-      }
-      catch (NexusClientAccessForbiddenException e) {
-        afterUploadFailure(request.getParameters(), remoteNexus, zappedStagingRepositories, e);
-        log.error("Remote staging finished with a failure: {}", e.getMessage());
-        log.error("");
-        log.error("Possible causes of 403 Forbidden:");
-        log.error(
-            " * you have no permissions to stage against profile with ID \"{}\"? Get to Nexus admin...", profileId);
-        throw new ArtifactDeploymentException("Remote staging failed: " + e.getMessage(), e);
+      String nexusUrl = readNexusUrl(nexusUrlFolder);
+      request.getParameters().setNexusUrl(nexusUrl); // All nexus staging configurations are reused except nexus url
+      RemoteNexus remoteNexus = createRemoteNexus(request.getMavenSession(), request.getParameters());
```

Comment:
Inherit parameters from last maven module except nexus url. Better to clone parameters instead of update it whenever wanted.